src/test/java/com/google/gcp_variant_transforms/library/HeaderIteratorTest.java

```diff
@@ -0,0 +1,145 @@
+    for (int i = 0; i < listSize; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    headerLines = headerLinesBuilder.build();
+    headerIterator = new HeaderIterator(headerLines);
+    // Act
+    for (int i = 0; i < listSize; i++) {
+      headerIterator.next();
+    }
+    // Assert
+    assertThat(headerIterator.hasNext()).isFalse();
+  }
+
+  @Test
+  public void testHeaderLinesHasNext_whenEmptyCheckingBoolean_thenFalse() {
+    // Arrange
+    int listSize = 5;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    ImmutableList<String> headerLines; 
+    HeaderIterator headerIterator;
+    headerLines = headerLinesBuilder.build(); // empty headerList
+    headerIterator = new HeaderIterator(headerLines);
+    // Act
+
+    // Assert
+    assertThat(headerIterator.hasNext()).isFalse();
+  }
+
+  @Test
+  public void testHeaderLinesNext_whenCheckingString_thenContains() {
+    // Arrange
+    int listSize = 5;
+    CharSequence expectedHeaderLine; 
+    ImmutableList<String> headerLines;
+    HeaderIterator headerIterator;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    for (int i = 0; i < listSize; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    headerLines = headerLinesBuilder.build();
+    headerIterator = new HeaderIterator(headerLines);
+    expectedHeaderLine = "##sampleHeaderLine=0";
+    // Act
+
+    // Assert
+    assertThat(headerIterator.next()).contains(expectedHeaderLine);
+  }
+
+  @Test
+  public void testHeaderLinesNext_whenCheckingString_thenException() {
+    // Arrange
+    int listSize = 5;
+    HeaderIterator headerIterator;
+    ImmutableList<String> headerLines;
+    Exception actualException;
+    Class expectedException = IndexOutOfBoundsException.class;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    headerLines = headerLinesBuilder.build(); // empty headerList
+    headerIterator = new HeaderIterator(headerLines);
+    // Act
+    actualException = assertThrows(IndexOutOfBoundsException.class, () ->
+        headerIterator.next());
+    // Assert
+    assertThat(actualException).isInstanceOf(expectedException);
+  }
+
+  @Test
+  public void testHeaderLinesPeek_whenCheckingString_thenContains() {
+    // Arrange
+    int listSize = 5;
+    CharSequence expectedHeaderLine; 
+    ImmutableList<String> headerLines;
+    HeaderIterator headerIterator;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    for (int i = 0; i < listSize; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    headerLines = headerLinesBuilder.build();
+    headerIterator = new HeaderIterator(headerLines);
+    expectedHeaderLine = "##sampleHeaderLine=0";
+    // Act
+
+    // Assert
+    assertThat(headerIterator.peek()).contains(expectedHeaderLine);
```

Comment:
Lets duplicate this line, to assert that peek doesn't move the pointer.

assertThat(headerIterator.peek()).contains(expectedHeaderLine);
assertThat(headerIterator.peek()).contains(expectedHeaderLine);

Reply:
I wasn't sure if I could have multiple asserts in one test. I agree, this is an important check to include.
resolved